src/cls/tabular/fbwriter_fbu.cc

```diff
@@ -0,0 +1,932 @@
+    // build out FB_Meta
+    // --------------------------------------------- //
+    flatbuffers::FlatBufferBuilder *meta_builder = new flatbuffers::FlatBufferBuilder();
+    Tables::createFbMeta( meta_builder, 
+                          SFT_FLATBUF_UNION_COL,
+                          reinterpret_cast<unsigned char*>( bl_seq.c_str() ),
+                          bl_seq.length() ) ;
+
+    // add meta_builder's data into a bufferlist as char*
+    ceph::bufferlist meta_bl ;
+    char* meta_builder_ptr = reinterpret_cast<char*>( meta_builder->GetBufferPointer() ) ;
+    int meta_builder_size  = meta_builder->GetSize() ;
+    std::cout << "meta_builder_size = " << meta_builder_size << std::endl ;
+    meta_bl.append( meta_builder_ptr, meta_builder_size ) ;
+    delete meta_builder;
+    librados::bufferlist meta_wrapper_bl ;
+    ::encode( meta_bl, meta_wrapper_bl ) ;
+    size_t meta_wrapper_bl_sz = meta_wrapper_bl.length() ;
+    std::cout << "meta_wrapper_bl_sz = " << meta_wrapper_bl_sz << std::endl ;
+
+    // --------------------------------------------- //
+    // do the write
+    // --------------------------------------------- //
+    // write bufferlist
+    if( inputs.writeto == "ceph" )
+      //writeToCeph( bl_seq, buffer_size, inputs.targetformat, inputs.targetoid, inputs.targetpool ) ;
+      writeToCeph( meta_wrapper_bl, meta_builder_size, inputs.targetformat, inputs.targetoid, inputs.targetpool ) ;
+    else if( inputs.writeto == "disk" )
+      writeToDisk( bl_seq, buffer_size, inputs.targetformat, inputs.targetoid ) ;
+    else {
+      std::cout << ">>> unrecognized writeto '" << inputs.writeto << "'" << std::endl ;
+      exit( 1 ) ;
+    }
+
+  } //cols
+
+  // otherwise oops
+  else {
+    std::cout << ">> unrecognized inputs.write_type = '" << inputs.write_type << "'" << std::endl ;
+  }
+} // do_write
+
+std::string getSchemaString( uint64_t ncols, 
+                             std::vector< std::string > schema_attnames, 
+                             std::vector< Tables::SkyDataType > schema_datatypes_sdt,
+                             bool debug ) {
+  std::string schema_string = "" ;
+  for( unsigned int i = 0; i < ncols; i++ ) {
+    std::string att_name = schema_attnames[ i ] ;
+    auto att_type = schema_datatypes_sdt[ i ] ;
+    std::string this_entry = " " + std::to_string( i ) + " " ;
+    switch( att_type ) {
+      case Tables::SDT_UINT64 :
+        this_entry = this_entry + std::to_string( Tables::SDT_UINT64 ) ;
+        break ;
+      case Tables::SDT_FLOAT :
+        this_entry = this_entry + std::to_string( Tables::SDT_FLOAT ) ;
+        break ;
+      case Tables::SDT_STRING :
+        this_entry = this_entry + std::to_string( Tables::SDT_STRING ) ;
+        break ;
+      default :
+        std::cout << ">>4 unrecognized att_type '" << att_type << "'" << std::endl ;
+        exit( 1 ) ;
+    } //switch
+    this_entry = this_entry + " 0 0 " + att_name + " \n" ;
```

Comment:
can you instead get the " 0 0 " values from the input schema or const schema string?  since we may want some to be keys or nullable like lineitem example schema https://github.com/uccross/skyhookdm-ceph/blob/skyhook-luminous/src/cls/tabular/cls_tabular_utils.h#L693

Reply:
parameterized the is-key and is-nullable values. they're now passed in at the command line like the list of data types and att names.

Reply:
great